src/platforms/shared.rs

```diff
@@ -22,7 +22,6 @@ pub const SHARED_BUILD_OPTS: [&str; 19] = [
     "-DLLVM_ENABLE_TERMINFO='Off'",
     "-DLLVM_ENABLE_LIBEDIT='Off'",
     "-DLLVM_ENABLE_LIBPFM='Off'",
-    "-DLLVM_ENABLE_ASSERTIONS='On'",
```

Comment:
@hedgar2017 , it will be already [default behavior](https://llvm.org/docs/CMake.html#:~:text=LLVM_ENABLE_ASSERTIONS%3ABOOL,is%20Debug.).

Reply:
Just to clarify the PR title, we're disabling assertions for all platforms, aren't we?

Reply:
> Still more convenient to have it here instead of browsing the docs.
Do you still prefer to add an option for that? I believe we should keep the default behavior and not change it.

Reply:
> Just to clarify the PR title, we're disabling assertions for all platforms, aren't we?
Yeah, good catch - updated.

Reply:
@antonbaliasnikov it was more like Flash's suggestion, but IIRC assertions can be super helpful for debugging, so keeping them in the debug makes sense.
There is already `build_type: BuildType` in every function so it's easy to add.

Reply:
@hedgar2017 , the `build_type` exists only for the target builds, for `stage1` host and `compiler-rt` you're always using `Release`. Do you want to add `build_type` support for these ones too?

Reply:
@antonbaliasnikov no, assertions are only needed for the last stage, as that is one LLVM squad develops and needs to debug.

Reply:
Could you add that failed target to the CI here please?

Reply:
Should be OK now, added everything.

Btw, with this build setup will not be possible to reproduce [CPR-1592](https://linear.app/matterlabs/issue/CPR-1592/linux-musl-llvm-targets-cannot-be-built-with-assertions), because this assertion was in the stage1 compiler.